python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -344,15 +283,9 @@ def predict_model_on_cpu(self, X, convert_dtype):
                 )
             )
 
-        wait(futures)
-        raise_exception_from_futures(futures)
-
-        indexes = list()
-        rslts = list()
-        for d in range(len(futures)):
-            rslts.append(futures[d].result())
-            indexes.append(0)
+        wait_and_raise_from_futures(futures)
 
+        rslts = [pred_rslts.result() for pred_rslts in futures]
```

Comment:
`results = self.client.gather(futures)`  should do this in the most efficient way. 
Actually, I think you can replace `wait_and_raise` plus line 288 with just:

`results = self.client.gather(futures, errors="raise")`